src/composables/useNotificationSystem.ts

```diff
@@ -1,43 +1,149 @@
      */
-    const addNotification = (notification: Notification) => {
-        notificationQueue.value.push(notification);
+    const createNotification: CreateNotification = (options) => {
+        const _options = Object.assign(
+            { ...defaultNotificationOptions },
+            options
+        );
+
+        notifications.value.push(
+            ...[
+                {
+                    id: createUUID(),
+                    ..._options,
+                },
+            ]
+        );
     };
 
     /**
-     * Function that can be used to remove a Notification from the queue.
-     * @param notification The notification to remove from the queue.
+     * Create an error notification
+     * @param options
      */
-    const removeNotification = (notification: Notification) => {
-        const index = notificationQueue.value.indexOf(notification);
-        if (index > -1) {
-            notificationQueue.value.splice(index, 1);
-        }
+    const createErrorNotification: CreateNotification = (options) => {
+        createNotification({
+            type: 'error',
+            title: 'Yikes. Something went wrong.',
+            duration: 8,
+            ...options,
+        });
+    };
+
+    /**
+     * Create a success notification
+     * @param options
+     */
+    const createSuccessNotification: CreateNotification = (options) => {
+        createNotification({ type: 'success', title: 'Success!', ...options });
+    };
+
+    /**
+     * Create a warning notification
+     * @param options
+     */
+    const createWarningNotification: CreateNotification = (options) => {
+        createNotification({
+            type: 'warning',
+            title: 'Something to lookout for.',
+            duration: 8,
+            ...options,
+        });
+    };
+
+    /**
+     * Remove notification from the list
+     * @param id
+     */
+    const removeNotifications = (id: string) => {
+        const index = notifications.value.findIndex((item) => item.id === id);
+        if (index !== -1) notifications.value.splice(index, 1);
+    };
+
+    /**
+     * Stop body overflow when notification is shown
+     */
+    const stopBodyOverflow = () => {
+        document && document.body.classList.add(...['hide-overflow']);
+    };
+
+    /**
+     * Allow body overflow after notification is removed
+     */
+    const allowBodyOverflow = () => {
+        document && document.body.classList.remove(...['hide-overflow']);
     };
 
     return {
-        addNotification,
-        removeNotification,
-        notificationQueue,
+        notifications,
+        createNotification,
+        createSuccessNotification,
+        createErrorNotification,
+        createWarningNotification,
+        removeNotifications,
+        stopBodyOverflow,
+        allowBodyOverflow,
     };
-};
+}
 
 /**
- * Interface that can be used to define a warning.
+ * Create a unique id
+ * @returns a unique id as a string
+ */
+function createUUID(): string {
```

Comment:
This function is redundant to generateRandomToken in App.vue (whyever it's in App.vue).
App.vue needs to be refactored first (who the hell implemented the entire session settings in there??)

Reply:
This is part of the notification, for a unique id for each notification, to identify it

Reply:
Yes, it generates a random string. Totally redundant.

Reply:
The current system uses this, as unique Key here. If we remove that bugs will occour.
<img width="404" alt="image" src="https://github.com/VisualGameData/VIGAD/assets/70775490/17fe736d-82d4-4e5c-abf6-92f409e006cd">

And the current implementation uses it to identify want element to remove
<img width="692" alt="image" src="https://github.com/VisualGameData/VIGAD/assets/70775490/b38c693f-49ab-4e2f-ba51-8a71dd7c6831">

Reply:
I understand that, that is still redundant. You don't need two methods to generate a random string (which a UUID essentially is). We should have one external helper composable that implements functions like this